### ISSUE TITLE
Does fabric-load-gen do the job of signing tx?

### ISSUE THREAD

**Issue by GingerMoon** [thakkarparth007/fabric-load-gen#1]
Dear thakkarparth007, 

It seems that the signing the tx via the client fabric-sdk takes so much time that only one host  cannot trigger large numbers of txs in short time and hence the benchmark cannot be done one only one host.

How does fabric-load-gen trigger a large number of transactions in a short time?
Does the client side still signing the txs? Or does it use "cold start", which means read the serialized endorsed tx directly from disk and send them to orderer directly?

Best Regards,
Xiang

**Comment by thakkarparth007** [comment]
Hi Xiang

So for the paper, we used a 32 core machine for the client, and it was able to generate 1.5k+ tps easily. For higher loads, we used 3 such machines, if I remember correctly.

Are you sure you're using correct `numProcesses` in the config?
There was a bottleneck where we used to verify the endorsement signatures received by the client, but we removed that check simply for improving load (without affecting correctness, since we always had correct endorsement signatures).

Regarding your second question, no we do not use a cold start method. It's an interesting approach, but I'm not sure I'd go as far as having serialized endorsed tx directly. That'd not let us test how endorserment phase performs. What I feel can be done instead is to have the client create serialized tx _proposals_ and send them to the endorsers, saving it the signature generation time.
